src/cli/self_update/test.rs

```diff
@@ -37,37 +42,35 @@ pub fn with_saved_path(f: &mut dyn FnMut()) {
 }
 
 #[cfg(windows)]
-pub fn get_path() -> io::Result<Option<RegValue>> {
-    get_reg_value(&RegKey::predef(HKEY_CURRENT_USER), "Environment", "PATH")
+pub fn get_path() -> io::Result<Option<String>> {
+    get_reg_string(CURRENT_USER, "Environment", "PATH")
 }
 
 #[cfg(windows)]
-pub fn with_saved_reg_value(root: &RegKey, subkey: &str, name: &str, f: &mut dyn FnMut()) {
+pub fn with_saved_reg_string(root: &Key, subkey: &str, name: &str, f: &mut dyn FnMut()) {
     with_saved_global_state(
-        || get_reg_value(root, subkey, name),
-        |p| restore_reg_value(root, subkey, name, p),
+        || get_reg_string(root, subkey, name),
+        |p| restore_reg_string(root, subkey, name, p),
         f,
     )
 }
 
 #[cfg(windows)]
-fn get_reg_value(root: &RegKey, subkey: &str, name: &str) -> io::Result<Option<RegValue>> {
-    let subkey = root.open_subkey_with_flags(subkey, KEY_READ | KEY_WRITE)?;
-    match subkey.get_raw_value(name) {
+fn get_reg_string(root: &Key, subkey: &str, name: &str) -> io::Result<Option<String>> {
+    let subkey = root.create(subkey)?;
+    match subkey.get_string(name) {
         Ok(val) => Ok(Some(val)),
-        Err(ref e) if e.kind() == io::ErrorKind::NotFound => Ok(None),
-        Err(e) => Err(e),
+        Err(ref e) if e.code() == HRESULT::from_win32(ERROR_FILE_NOT_FOUND) => Ok(None),
```

Comment:
Are we sure this is the best way of expressing the condition? Can't we look at the error code in `e` "directly" somehow?

Reply:
I think we can't, because `Error` struct in `windows-result` doesn't hold the error code directly. Plus, the previous code also didn't expose its error code, so I made this change.

Reply:
@djc HRESULT includes both Win32 and WinNT error codes, this `from_win32` is just a arithmetic mapping function without much overhead.